shadowsocks/tcp.go

```diff
@@ -106,24 +106,35 @@ func findAccessKey(clientConn onet.DuplexConn, cipherList CipherList) (string, o
 		}
 		// Move the active cipher to the front, so that the search is quicker next time.
 		cipherList.SafeMarkUsedByClientIP(entry, clientIP)
-		ssr := NewShadowsocksReader(io.MultiReader(bytes.NewReader(replayBytes), clientConn), cipher)
+		ssr := NewShadowsocksReader(io.MultiReader(bytes.NewReader(firstBytes), clientConn), cipher)
 		ssw := NewShadowsocksWriter(clientConn, cipher)
-		return id, onet.WrapConn(clientConn, ssr, ssw).(onet.DuplexConn), nil
+		return id, onet.WrapConn(clientConn, ssr, ssw).(onet.DuplexConn), salt, nil
 	}
-	return "", clientConn, fmt.Errorf("Could not find valid TCP cipher")
+	return "", clientConn, firstBytes, fmt.Errorf("Could not find valid TCP cipher")
```

Comment:
Let's return the `salt` instead. The `firstBytes` is not that meaningful.

Reply:
`salt` doesn't exist here, because we haven't selected a cipher, so we don't know the salt size.  I changed this to return `nil` instead.